### ISSUE TITLE
Logout broken in Symfony 3.4.10

### ISSUE THREAD

**Issue by nreynis** [hslavich/OneloginSamlBundle#53]
During logout, when the IDP redirect to `/saml/logout` on the SP, an AccessDeniedException is now thrown.
Which result in the user being redirected to the IDP's login page instead of the logout target specified in `security.yml`.

I think (but I'm not totally sure) that this has been introduced by changes made on the default logout_listener in Symfony 3.4.10:
https://symfony.com/blog/symfony-3-4-10-released
https://github.com/symfony/symfony/pull/24805

I have fixed the issue by making `/saml/logout` public:
```yaml
security:
    access_control:
        - { path: ^/saml/(login|logout|metadata), roles: IS_AUTHENTICATED_ANONYMOUSLY }
```

I'm not sure this is the best way to fix it though. I do not have enough knowledge of the inner mechanics of Symfony's SecurityBundle.

**Comment by xelan** [comment]
Hi nreynis!

When investigating on a related problem, probably with the same cause, I stumbled across your issue here. 

I was able to reproduce the problem with Symfony v2.7.47 and v3.4.10. Probably other versions where PR 24805 was merged are affected as well. Therefore, I opened [an issue](https://github.com/symfony/symfony/issues/28058) at the symfony project.

**Comment by nreynis** [comment]
Thank you for the heads-up, I wasn't sure this was a symfony issue or a legitimate change.
I'm monitoring the subject.

**Comment by hslavich** [comment]
Is this fixed?

**Comment by nreynis** [comment]
On 4.x it is. I don't have the time to check the 3.x branch, but it's likely fixed too. Can someone confirm?

**Comment by xelan** [comment]
Yes, it is fixed on all supported Symfony versions (2.8+)
